src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildStep.java

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.datatypes;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+
+/**
+ * Contains the information for an individual step of a build.
+ */
+@Entity(name = "buildStep")
+public class BuildStep {
+
+  // Order of the step
+  @Field(name="step_number")
+  private int stepNumber;
+
+  // BuildStep name (e.g "clean-src-dir")
+  @Field(name="name")
+  private String name;
+
+  // BuildStep text (e.g "clean-src-dir skipped")
+  @Field(name="text")
+  private String text;
+
+  // Indicator of whether or not the step is complete
+  @Field(name="isFinished")
+  private boolean isFinished;
+
+  // Indicator of whether or not the step has started
+  @Field(name="isStarted")
+  private boolean isStarted;
+
+  // All the logs associated with this step
+  @Field(name="logs")
+  private List<Log> logs;
+
+  /**
+   * Name, text and logs can be null.
+   * @param name - name of the build step.
+   * @param text - output text related to step's build status.
+   * @param isFinished - indicates if the step has been completed.
+   * @param isStarted - indicates if the step has been initiated.
+   * @param logs - the logs associated with the build step.
+   */
+  BuildStep(String name, String[] text, boolean isFinished, boolean isStarted, Log[] logs) {
```

Comment:
2 things:
- It's confusing that the `text` parameter is an array, whereas the text member is just String. I'd suggest turning the parameter into a `String`; callers that have an array can do the Arrays conversion by themselves, if needed.
- For consistency with other datatypes here, use an `List<Log>` instead of a `Log[]` for the `logs` parameter; see for example the BuildBotData's constructor's buildSteps parameter.